pkg/updatestrategy/aws_ec2.go

```diff
@@ -203,7 +204,17 @@ func (n *EC2NodePoolBackend) Scale(context.Context, *api.NodePool, int) error {
 	return nil
 }
 
-func (n *EC2NodePoolBackend) Terminate(context.Context, *Node, bool) error {
+func (n *EC2NodePoolBackend) Terminate(ctx context.Context, pool *api.NodePool, node *Node, _ bool) error {
+	// terminating the instance using AWS api, it will also trigger karpenter interruption controller to
+	// delete the node and nodeClaim objects
+	instanceID := instanceIDFromProviderID(node.ProviderID, node.FailureDomain)
+	params := &ec2.TerminateInstancesInput{
+		InstanceIds: []*string{&instanceID},
+	}
+	_, err := n.ec2Client.TerminateInstancesWithContext(ctx, params)
+	if err != nil {
+		return fmt.Errorf("failed to terminate EC2 instances of the node pool '%s': %w", render.Render(pool), err)
```

Comment:
Don't we have the node pool name on the `*Node` object? It seems a bit much to change the interface and pass `pool *api.NodePool` everywhere and then only use it in this error log :)

Reply:
it is not used for this one only. the `ProfileNodePoolProvisioner` rely on the Profile to select the right backend (karpenter vs ASG). otherwise it defaults to ASG. it does not appear in the diff because ProfileNodePoolProvisioner was already used, since it was lacking the nodePool it was defaulting to ASG.

I can try to simplify the interface